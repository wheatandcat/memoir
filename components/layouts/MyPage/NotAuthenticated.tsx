import Button from '@/components/elements/Button';
import Text from '@/components/elements/Text';
import View from '@/components/elements/View';
import UserImage from '@/components/layouts/User/Image';
import { MaterialIcons } from '@expo/vector-icons';
import type { ConnectedType } from 'components/pages/MyPage/Connected';
import theme from 'config/theme';
import { type FC, memo } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import type { User } from 'store/atoms';

export type Props = {
  user: User;
  onLogin: () => void;
} & Pick<ConnectedType, 'onUpdateProfile'>;

const NotAuthenticated: FC<Props> = (props) => {
  return (
    <View style={styles.root}>
      <View style={styles.user}>
        <View mt={4}>
          <TouchableOpacity onPress={props.onUpdateProfile}>
            <UserImage image={props.user.image} />
          </TouchableOpacity>
        </View>
        <View m={4}>
          <TouchableOpacity onPress={props.onUpdateProfile}>
            <View style={styles.userInfo}>
              <Text>
                {props.user?.displayName || '表示名の設定がありません'}
              </Text>
              <MaterialIcons
                name="chevron-right"
                size={23}
                color={theme().color.secondary.main}
              />
            </View>
          </TouchableOpacity>
        </View>
      </View>

      <View m={3} style={styles.action}>
        <View my={4} style={styles.text}>
          <Text variants="small" textAlign="center" lineHeight={15}>
            サインインすれば、他のユーザーとタスクを共有することができます。
          </Text>
        </View>
        <View>
          <Button title="サインイン" onPress={props.onLogin} width={220} />
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    height: '100%',
    alignItems: 'center',
  },
  user: {
    width: '100%',
    alignItems: 'center',
  },
  userInfo: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  action: {
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    width: 250,
  },
});

export default memo(NotAuthenticated);
