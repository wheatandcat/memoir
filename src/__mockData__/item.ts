import { ItemQuery } from 'queries/api/index';
type Item = NonNullable<ItemQuery['item']>;

export const item = (option?: Partial<Item>): Item => ({
  id: '1',
  title: '買い物',
  categoryID: 9,
  date: '2021-01-01T00:00:00+09:00',
  like: false,
  dislike: false,
  createdAt: '2021-01-01T00:00:00+09:00',
  updatedAt: '2021-01-01T00:00:00+09:00',
  ...option,
});

export const items = (): Item[] => [
  item(),
  item({
    id: '2',
    title: '本を読む',
    categoryID: 2,
    date: '',
    dislike: false,
    like: false,
  }),
];
