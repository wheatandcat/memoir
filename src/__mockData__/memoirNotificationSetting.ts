import { State as MemoirNotificationSettingState } from 'hooks/useMemoirNotificationSetting';

export const memoirNotificationSetting = (
  option?: Partial<MemoirNotificationSettingState>
): MemoirNotificationSettingState => ({
  dayOfWeek: 0,
  hours: 0,
  minutes: 0,
  notification: false,
  ...option,
});
