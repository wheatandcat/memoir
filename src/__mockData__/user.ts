import { User } from 'queries/api/index';

export const user = (option?: Partial<User>): User => ({
  id: '1',
  displayName: 'test',
  image: '',
  updatedAt: '2021-01-01T00:00:00+09:00',
  createdAt: '2021-01-01T00:00:00+09:00',
  ...option,
});

export const users = (): User[] => [
  {
    id: 'test1',
    displayName: 'suzuki1',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test2',
    displayName: 'suzuki2',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test3',
    displayName: 'suzuki4',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test4',
    displayName: 'suzuki4',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
  {
    id: 'test5',
    displayName: 'suzuki5',
    image: 'https://placehold.jp/150x150.png',
    updatedAt: '2021-01-01T00:00:00+09:00',
    createdAt: '2021-01-01T00:00:00+09:00',
  },
];
