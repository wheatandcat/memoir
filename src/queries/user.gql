query User {
  user {
    id
    displayName
  }
}
