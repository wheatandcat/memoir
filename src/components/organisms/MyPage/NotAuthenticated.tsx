import React, { memo } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import theme from 'config/theme';
import View from 'components/atoms/View';
import Button from 'components/atoms/Button';
import UserImage from 'components/molecules/User/Image';
import Text from 'components/atoms/Text';
import { User } from 'store/atoms';
import { ConnectedType } from 'components/pages/MyPage/Connected';
import { MaterialIcons } from '@expo/vector-icons';

export type Props = {
  user: User;
  onLogin: () => void;
} & Pick<ConnectedType, 'onUpdateProfile'>;

const NotAuthenticated: React.FC<Props> = (props) => {
  return (
    <View style={styles.root}>
      <View style={styles.user}>
        <View mt={4}>
          <UserImage image={props.user.image} />
        </View>
        <View m={4}>
          <TouchableOpacity onPress={props.onUpdateProfile}>
            <View style={styles.userInfo}>
              <Text>
                {props.user?.displayName || '表示名の設定がありません'}
              </Text>
              <MaterialIcons
                name="chevron-right"
                size={23}
                color={theme().color.secondary.main}
              />
            </View>
          </TouchableOpacity>
        </View>
      </View>

      <View m={3} style={styles.action}>
        <View my={4} style={styles.text}>
          <Text variants="small" textAlign="center" lineHeight={15}>
            サインインすれば、他のユーザーとタスクを共有するこができます
          </Text>
        </View>
        <View>
          <Button title="サインイン" onPress={props.onLogin} width={220} />
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    height: '100%',
    alignItems: 'center',
  },
  user: {
    width: '100%',
    alignItems: 'center',
  },
  userInfo: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  action: {
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    width: 250,
  },
});

export default memo(NotAuthenticated);
