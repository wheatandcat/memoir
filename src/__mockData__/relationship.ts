import type { RelationshipsQuery } from "queries/api/index";
import { user } from "./user";

export type Relationship = NonNullable<
  EdgesNode<RelationshipsQuery["relationships"]>
>;

export const relationship = (option?: Partial<Relationship>): Relationship => ({
  id: "1",
  followedId: "test",
  followerId: "test",
  user: user(),
  createdAt: "2021-01-01T00:00:00+09:00",
  updatedAt: "2021-01-01T00:00:00+09:00",
  ...option,
});

export const relationships = (): Relationship[] => [
  relationship(),
  relationship({
    id: "2",
    user: user(),
  }),
  relationship({
    id: "3",
    user: user(),
  }),
  relationship({
    id: "4",
    user: user(),
  }),
];
