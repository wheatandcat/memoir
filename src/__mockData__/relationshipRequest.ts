import { RelationshipRequestsQuery } from 'queries/api/index';
import { user } from './user';

export type RelationshipRequest = NonNullable<
  EdgesNode<RelationshipRequestsQuery['relationshipRequests']>
>;

export const relationshipRequest = (
  option?: Partial<RelationshipRequest>
): RelationshipRequest => ({
  id: '1',
  followedId: 'test',
  followerId: 'test',
  user: user(),
  createdAt: '2021-01-01T00:00:00+09:00',
  updatedAt: '2021-01-01T00:00:00+09:00',
  ...option,
});

export const relationshipRequests = (): RelationshipRequest[] => [
  relationshipRequest(),
  relationshipRequest({
    id: '2',
    user: user(),
  }),
  relationshipRequest({
    id: '3',
    user: user(),
  }),
  relationshipRequest({
    id: '4',
    user: user(),
  }),
];
