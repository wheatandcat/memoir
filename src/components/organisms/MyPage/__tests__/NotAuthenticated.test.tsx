import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { user } from '__mockData__/user';
import NotAuthenticated, { Props } from '../NotAuthenticated';

const propsData = (): Props => ({
  user: {
    ...user(),
    userID: '',
  },
  onLogin: jest.fn(),
  onUpdateProfile: jest.fn(),
});

describe('components/organisms/MyPage/NotAuthenticated.tsx', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<NotAuthenticated {...propsData()} />);
  });

  it('正常にrenderすること', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
