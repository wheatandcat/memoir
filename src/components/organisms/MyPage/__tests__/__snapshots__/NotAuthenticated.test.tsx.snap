// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/MyPage/NotAuthenticated.tsx 正常にrenderすること 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F2F2F2",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "width": "100%",
      }
    }
  >
    <View
      mt={4}
    >
      <ForwardRef
        onPress={[MockFunction]}
      >
        <Memo(UserImage)
          image=""
        />
      </ForwardRef>
    </View>
    <View
      m={4}
    >
      <ForwardRef
        onPress={[MockFunction]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            fontFamily="RobotoCondensed-Bold"
            variants="body"
          >
            test
          </Text>
          <Icon
            allowFontScaling={false}
            color="#362740"
            name="chevron-right"
            size={23}
          />
        </View>
      </ForwardRef>
    </View>
  </View>
  <View
    m={3}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      my={4}
      style={
        Object {
          "width": 250,
        }
      }
    >
      <Text
        fontFamily="RobotoCondensed-Bold"
        lineHeight={15}
        textAlign="center"
        variants="small"
      >
        サインインすれば、他のユーザーとタスクを共有することができます。
      </Text>
    </View>
    <View>
      <Button
        fontFamily="NotoSansJP-Bold"
        onPress={[MockFunction]}
        size="base"
        title="サインイン"
        width={220}
      />
    </View>
  </View>
</View>
`;
